### ISSUE TITLE
consider using a binary format instead of json

### ISSUE THREAD

**Issue by runcom** [containers/storage#122]
Not sure the various json files must be user friendly so take this lightly. But, as since is super slow (and requires hacks like #121) I would love to switch to something binary-ish.

@rhatdan @nalind is this something we can consider? This will speed things up for sure.

**Comment by abitrolly** [comment]
@runcom where do you expect the most performance improvements from this change?

**Comment by abitrolly** [comment]
I've just noticed that `containers.json` in `vfs` is minified. What is the performance penalty of storing it in readable indented format? Size is unlikely to the reason for dir that is measured in GB.

**Comment by rhatdan** [comment]
@nalind Do you still consider this an issue.  BTW libpod moved to a different json library should we move containers/storage to the same library?

**Comment by rhatdan** [comment]
@mheon Should we move to the same image format you are using in libpod?

**Comment by mheon** [comment]
The problem with using `jsoniterator` here is compatability. `ffjson` does not serialize certain things in the same way that the Go standard library (and `jsoniterator` and `easyjson`) do - most notably for us was IP addresses, which serialized as strings for everyone except `ffjson`, which base64's the raw bytes. So we may encounter compatability issues in attempting to migrate.

This is actually why `libpod` quickly moved off of `ffjson` - we did not use it at first, and when we migrated, certain fields didn't unmarshal properly, and we ended up with unusable, unremovable containers until we quickly migrated to an alternative that was compatible with the standard library.
